model/generate.go

```diff
@@ -142,35 +141,38 @@ func (g *GeneratedProject) NewVersion() (*Project, *Version, *task.Task, *projec
 
 	// Validate generated project against original project.
 	if err = g.validateGeneratedProject(p, cachedProject); err != nil {
-		return nil, nil, nil, nil, "",
+		return nil, nil, nil, nil,
 			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrap(err, "generated project is invalid").Error()}
 	}
 
-	config, err := g.addGeneratedProjectToConfig(v.Config, cachedProject)
+	newConfig, err := g.addGeneratedProjectToConfig(v.Config, cachedProject)
 	if err != nil {
-		return nil, nil, nil, nil, "",
+		return nil, nil, nil, nil,
 			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrap(err, "error creating config from generated config").Error()}
 	}
-	v.Config = config
+	v.Config = newConfig
 	if err := LoadProjectInto([]byte(v.Config), t.Project, p); err != nil {
-		return nil, nil, nil, nil, "",
+		return nil, nil, nil, nil,
 			gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: errors.Wrap(err, "error reading project yaml").Error()}
 	}
-	return p, v, t, &cachedProject, prevConfig, nil
+	return p, v, t, &cachedProject, nil
 }
 
-func (g *GeneratedProject) Save(ctx context.Context, p *Project, v *Version, t *task.Task, pm *projectMaps, prevConfig string) error {
+func (g *GeneratedProject) Save(ctx context.Context, p *Project, v *Version, t *task.Task, pm *projectMaps) error {
 	query := bson.M{
-		VersionIdKey:     v.Id,
-		VersionConfigKey: prevConfig,
+		VersionIdKey:           v.Id,
+		VersionConfigNumberKey: v.ConfigUpdateNumber,
+	}
+	update := bson.M{
+		"$set": bson.M{VersionConfigKey: v.Config},
+		"$inc": bson.M{VersionConfigNumberKey: 1},
 	}
-	update := bson.M{"$set": bson.M{VersionConfigKey: v.Config}}
 
 	err := VersionUpdateOne(query, update)
 	if err != nil {
 		return errors.Wrapf(err, "error updating version %s", v.Id)
 	}
-
+	v.ConfigUpdateNumber += 1
```

Comment:
I'm not sure that this is necessary. The version number will have been updated in the database above. Does `saveNewBuildsAndTasks` use the updated version number?

Reply:
Hmm no, I'm just a little uncomfortable having a struct passed around with conflicting information--if someone ever does need it in the future, it would be unintuitive that the config string and the update number don't match. I can remove it if you think there's room for error with it?

Reply:
I think that's reasonable.